lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -310,6 +310,8 @@ def from_string(cls, collection_input, artifacts_manager, supplemental_signature
     def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_options=True):
         req_name = collection_req.get('name', None)
         req_version = collection_req.get('version', '*')
+        if req_version is not None:
+            req_version = str(req_version)
```

Comment:
@s-hertel have you considered just doing this bare minimum?

Reply:
I don't think it's applying to all the potential matches in the provider, is it? Needs some tests, have no idea if/what issues this is fixing.

Reply:
Yeah, I know about the tests. I was just pulling some things from my `git stash` and turning them into draft PRs so they don't rot there.
This is an attempt to address the case when `requirements.yml` has a version that is parsed by PyYAML as an int or a float causing TypeError tracebacks when some logic down the line assumes the `ver` attribute is always a string, calling `str` methods.